tutorials/2_Gazebo_and_ROS.ipynb

```diff
@@ -0,0 +1,1330 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h2>Creating your Gazebo Launch File</h2>\n",
+    "\n",
+    "<p>Let’s start with the gazebo package, go in there and create the following subfolders:</p>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "roscd diff_wheeled_robot_gazebo \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "mkdir launch \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<p>Create a new file:</p>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "gedit diff_wheeled_gazebo.launch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<p>Add an empty world and try it </p>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<launch>\n",
+    "  <include file=\"$(find gazebo_ros)/launch/empty_world.launch\">\n",
+    "  </include>\n",
+    "</launch>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will include an empty world launch file from gazebo_ros package which will launch gazebo with nothing inside. Open a new terminal and launch the file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "roslaunch diff_wheeled_robot_gazebo diff_wheeled_gazebo.launch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<p>Now you should see the gazebo server and the gui starting with an empty world that contains a ground plane and a sun (which is not obviously visible without objects).</p>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h2>Creating your own Robot Model</h2>\n",
+    "\n",
+    "<p>First we need to go into our description package and create the urdf subfolder and the description file:</p>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "roscd diff_wheeled_robot_description"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "mkdir urdf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "cd urdf"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "gedit diff_wheeled_robot.xacro"
```

Comment:
Once again don't forget to edit this cell because gedit does not launch from inside a notebook like this.